### ISSUE TITLE
현재 데이터 요약 및 명세/정책 추가

### ISSUE THREAD

**Issue by iejob** [iejob/computer-vision-competition-1SEN#7]
## 1) 현재 데이터 요약(정량 근거 삽입 권고)
- `train.csv`: 1,570행, 컬럼 = `['ID','target']`
- `meta.csv`: 17행, 컬럼 = `['target','class_name']`
- `sample_submission.csv`: 3,140행, 컬럼 = `['ID','target']`
- 클래스 분포(정렬):
    
    `0:100, 1:46, 2:100, 3:100, 4:100, 5:100, 6:100, 7:100, 8:100, 9:100, 10:100, 11:100, 12:100, 13:74, 14:50, 15:100, 16:100`
    
- 권고 클래스 가중치(중앙값/빈도):
    
    ```
    {0:1.0, 1:2.1739, 2:1.0, 3:1.0, 4:1.0, 5:1.0, 6:1.0, 7:1.0,
     8:1.0, 9:1.0, 10:1.0, 11:1.0, 12:1.0, 13:1.3514, 14:2.0, 15:1.0, 16:1.0}
    ```
    

---

## 2) 구성/경로/파라미터 “단일 소스” 명세 추가
**목적:** 환경/경로/하이퍼를 한곳에서 관리하여 재현성·검증 용이성 확보.

**`data_config.json`(예시):**

```json
{
  "paths": {
    "train_csv": "data/raw/train.csv",
    "test_csv": "data/raw/sample_submission.csv",
    "meta_csv": "data/raw/meta.csv",
    "train_root": "data/raw/train",
    "test_root": "data/raw/test",
    "processed_dir": "data/processed",
    "reports_dir": "docs",
    "exp_dir": "experiments/data_loader"
  },
  "loader": {
    "img_size": 224,
    "num_workers": 4,
    "batch_size": 64,
    "pin_memory": true,
    "prefetch_factor": 2,
    "persistent_workers": true
  },
  "augs": {
    "rotate_limit": 25,
    "jitter": 0.2,
    "use_clahe": true,
    "use_blur": true,
    "use_affine": true,
    "hflip_p": 0.2
  },
  "sampler": {
    "use_weighted": true,
    "clip_min": 0.5,
    "clip_max": 3.0
  },
  "seed": 42,
  "strict_mode": true
}
```

> strict_mode: 무결성 위반 시 즉시 실패(머지 전 필수), false인 경우 대체 로드/스킵 허용(탐색 단계).

---

## 3) 실패 정책(Failure Policy) 명문화
문서에 아래와 같이 **엄격/관대한 동작**을 명시해 주십시오.

- **파일 누락/손상**
    - *Strict*: 프로세스 중단 + `data_issues.csv`에 기록 → 이슈 해결 후 재실행.
    - *Lenient*: 블랭크(혹은 평균 픽셀) 대체 이미지 반환 + 로그 경고(EDA/데모 전용).
- **CSV 스키마 불일치**: *항상 Strict* (머지 불가).
- **중복/누수 발견**: *항상 Strict* (머지 불가, 제거/마스킹 목록 확정 후 재검증).

---

## 4) 변환 파라미터 테이블(허용 범위·기본값)
문서 이미지 특성상 **가독성 훼손 방지**를 위해 허용 구간을 표로 고정하시길 권고합니다.

| 단계 | 파라미터 | 기본값 | 허용 범위 | 비고 |
| --- | --- | --- | --- | --- |
| Resize/Pad | size | 224 | {224,320,384} | Val/Test 동일 |
| Rotate | limit(°) | ±25 | ±15~±30 | 항상 p∈[0.5,0.8] |
| RandomResizedCrop | scale | (0.8,1.0) | (0.7,1.0) | ratio 0.9~1.1 |
| HorizontalFlip | p | 0.2 | 0.0~0.3 | 과도 flip 금지 |
| ColorJitter | strength | 0.2 | 0.1~0.3 | 밝기/대비/채도 |
| CLAHE | clip_limit | 2.0 | 1.0~3.0 | p≤0.3 권장 |
| GaussianBlur | k | 3~5 | 3~5 | p≤0.3 권장 |
| Affine | shear/scale | 소폭 | 제한적 | 기하 왜곡 최소화 |
| Normalize | mean/std | ImageNet | 고정 | timm 호환 |
| ToTensor |  |  |  | albumentations.pytorch |

> 표의 기본값을 코드 Default로 설정하고, 변경 시 config.json에만 반영되도록 통로를 단일화하십시오.

---

## 5) 유닛 테스트 파일(명시적 파일명) 추가
머지 조건에 포함될 **테스트 스위트**를 파일 단위로 고정하십시오.

- `tests/test_checks.py`
    - 스키마 검증: 필수 컬럼·라벨 도메인·결측 0
    - 파일 존재·0바이트·오픈 예외 0
- `tests/test_sampler.py`
    - 가중치 사전 산출(키 0~16, 값>0)
    - 10k 샘플 시뮬 분포 편차 ≤ ±8%p
- `tests/test_transforms.py`
    - (train) 배치 텐서 `dtype=float32`, 범위 ≈ 정규화
    - (val/test) 순수 전처리만 적용(증강 없음)
- `tests/test_splits.py`
    - StratifiedKFold 분포 근사 동일
    - Fold 간 ID·MD5 교집합 0

> CI가 없다면 최소한 로컬 pytest -q 통과 스크린샷을 PR에 첨부하도록 규정해 두는 것이 좋습니다.

---

## 6) 노트북 실행·내보내기 자동화(머지 체크의 일환)
- **실행**: `papermill` 또는 `jupyter nbconvert --to notebook --execute`로 00~05 일괄 실행
- **산출**: `01_transforms_gallery.ipynb`에서 `docs/eda_report.html|pdf` 내보내기
- **로그**: 각 노트북 최상단에 `run_id`/시드/환경(파이썬·패키지·CUDA) 요약 셀 필수

> “모든 셀 실행 완료” 조건이 머지 게이트로 자동 검증되도록 절차화하십시오.

---

## 7) DataLoader 성능 벤치마크 절차(수치 확정 방식)
문서에 **목표치(X imgs/s)**가 미정이라면, 다음 절차를 넣어 수치 합의→고정하시길 권고합니다.

1. 224px, `bs∈{32,64,128}`, `num_workers∈{2,4,8}` 전수 스윕
2. 각 조합 500배치 **워밍업 제외** 평균 imgs/sec 기록
3. 최고 조합의 **하위 10% 프랙타일** 값을 기준 삼아 `X` 확정(예: 320 imgs/s)
4. `timings.json`에 상세 기록(하드웨어/조합/분산 통계)

---

## 8) 로깅/감사 추적(감사 대응)
- 모든 체크 결과를 `experiments/data_loader/<run_id>/` 하위에 **JSON 로그**로 저장
- `data_issues.csv`(있을 시)와 함께 PR에 링크 첨부
- `strict_mode=true`에서 통과한 로그만 머지 허용

---

## 9) 비목표(Non-goals) 명시(스코프 크리프 방지)
- 텍스트 OCR/LLM 파이프라인 구성(규칙/스코프 외)
- Pseudo Labeling(규칙 위반)
- 외부 데이터/사전 라벨 확장(규칙 위반)
